mirrord/layer/src/socket/ops.rs

```diff
@@ -631,3 +634,42 @@ pub(super) fn getaddrinfo(
 
     Detour::Success(result)
 }
+
+fn remote_hostname_string() -> Detour<CString> {
+    let hostname_path = CString::new("/etc/hostname")?;
+
+    let hostname_fd = file::ops::open(
+        Some(&hostname_path),
+        OpenOptionsInternal {
+            read: true,
+            ..Default::default()
+        },
+    )?;
+
+    let hostname_file = file::ops::read(hostname_fd, 256)?;
+
+    close_layer_fd(hostname_fd);
+
+    CString::new(
+        hostname_file
+            .bytes
+            .into_iter()
+            .take(hostname_file.read_amount as usize - 1)
+            .collect::<Vec<_>>(),
+    )
+    .map(Detour::Success)?
+}
+
+/// Resolve the fake local address to the real local address.
+#[tracing::instrument(level = "trace")]
+pub(super) fn gethostname() -> Detour<&'static CString> {
+    let hostname = HOSTNAME.get_or_init(|| {
+        remote_hostname_string().ok().or_else(|| {
```

Comment:
Are we "hook guarded" here? Because in theory you could just do `std::fs::read_to_string("etc/hostname")` here.

Reply:
Also, if you're using the `else` case just to log, then you could use `inspect_err` before the `ok` conversion.

Reply:
that's cool idea